Building with "esModuleInterop": true on forces dependents to have it on too

`@types/react-native-dialog` depends on `react-native-modal`, but its tsconfig does not have `"esModuleInterop": true` on because Definitely Typed doesn't allow it: it's not on by default on old versions of Typescript.

Unfortunately, this means that dist/modal.d.ts has incorrect generated code:

```ts
import React from "react"
```

when, for backward compatibility, it should be

```ts
import React = require("react")
```

This affects any consumer of react-native-modal that doesn't have esModuleInterop on.

To reproduce, create a new typescript project with `tsc --init` and then `npm install react-native-modal`. Set `"esModuleInterop": false` and you will see a compile error:

```
node_modules/react-native-modal/dist/modal.d.ts:1:8 - error TS1259: Module '"/home/nathansa/DefinitelyTyped/types/react/index"' can only be default-imported using the 'esModuleInterop' flag

1 import React, { ReactNode } from 'react';
         ~~~~~

  ../react/index.d.ts:59:1
    59 export = React;
       ~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

node_modules/react-native-modal/dist/modal.d.ts:3:8 - error TS1192: Module '"/home/nathansa/DefinitelyTyped/types/prop-types/index"' has no default export.

3 import PropTypes from 'prop-types';
         ~~~~~~~~~
```

Here's an example for the nightly test run of Definitely Typed:

https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=22537&view=logs&j=b2dd9be2-fb32-52f8-c4c9-7dc0bc384587&t=f2dea62c-b696-5a51-78ad-206032e0a320&l=14604

--- a/src/modal.tsx
+++ b/src/modal.tsx
@@ -1,4 +1,5 @@
-import React, {ReactNode} from 'react';
+import * as React from 'react';
+import {ReactNode} from 'react';
 import {
   Animated,
   DeviceEventEmitter,

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,7 +9,7 @@
   "compilerOptions": {
     "lib": ["es5", "es6", "esnext.asynciterable"],
     "allowSyntheticDefaultImports": true,
-    "esModuleInterop": true,
+    "esModuleInterop": false,
     "jsx": "react",
     "moduleResolution": "node",
     "strict": true,

